Assets/Scripts/Enemy/EnemyView.cs

```diff
@@ -2,35 +2,44 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UI;
+using Interfaces;
+using System;
 
 namespace Enemy
 {
     [RequireComponent(typeof(Rigidbody))]
-    public class EnemyView : MonoBehaviour
+    public class EnemyView : MonoBehaviour, ITakeDamage
     {
 
+        public event Action increaseScore;
         private EnemyController enemyController;
 
-        public void SetEnemyController(EnemyController enemyController)
+        void Start()
         {
-            this.enemyController = enemyController;
+            enemyController.DestroyEnemy += DestroyEnemy;
         }
 
-        public void DamageEnemy(int value)
+        public void SetEnemyController(EnemyController enemyController)
         {
-            enemyController.TakeDamage(value);
+            this.enemyController = enemyController;
         }
 
-        public void DestroyEnemy()
+        private void DestroyEnemy()
         {
-            GameUI.Instance.UpdatePlayerScore(enemyController.enemyModel.scriptableObj.scoreIncrease);
+            Player.PlayerManager.Instance.playerController.setPlayerScore(enemyController.getScoreIncreaser());
```

Comment:
all this should be on the event - now your enemy view is hard coupled with player controller and enemy manager